crates/rattler_package_streaming/src/read.rs

```diff
@@ -86,3 +88,60 @@ pub fn extract_conda(reader: impl Read, destination: &Path) -> Result<ExtractRes
 
     Ok(ExtractResult { sha256, md5 })
 }
+
+/// Extracts a folder from a tar.bz2 archive.
+pub fn folder_from_tar_bz2(
+    archive_path: &Path,
+    find_path: &Path,
+    dest_folder: &Path,
+) -> Result<(), std::io::Error> {
+    let reader = std::fs::File::open(archive_path)?;
+    let mut archive = read::stream_tar_bz2(reader);
+
+    for entry in archive.entries()? {
+        let mut entry = entry?;
+        let path = entry.path()?;
+        if let Ok(stripped_path) = path.strip_prefix(find_path) {
+            let dest_file = dest_folder.join(stripped_path);
+            if let Some(parent_folder) = dest_file.parent() {
+                if !parent_folder.exists() {
+                    std::fs::create_dir_all(parent_folder)?;
+                }
+            }
+            let mut dest_file = std::fs::File::create(dest_file)?;
+            std::io::copy(&mut entry, &mut dest_file)?;
+        }
+    }
+    Ok(())
+}
+
+/// Extracts a folder from a conda archive.
+pub fn folder_from_conda(
+    archive_path: &Path,
+    find_path: &Path,
+    dest_folder: &Path,
+) -> Result<(), std::io::Error> {
+    let reader = std::fs::File::open(archive_path)?;
```

Comment:
Would it be beneficial to use a `BufReader` here? Im not sure if the zip library does buffering internally.